server/base/src/main/java/org/apache/accumulo/server/util/ZooZap.java

```diff
@@ -94,13 +94,15 @@ public void execute(String[] args) throws Exception {
       if (siteConf.getBoolean(Property.INSTANCE_RPC_SASL_ENABLED)) {
         SecurityUtil.serverLogin(siteConf);
       }
-      zap(siteConf, args);
+      try (var zk = new ZooSession(getClass().getSimpleName(), siteConf)) {
+        zap(zk, siteConf, args);
+      }
     } finally {
       SingletonManager.setMode(Mode.CLOSED);
     }
   }
 
-  public void zap(SiteConfiguration siteConf, String... args) {
+  public void zap(ZooSession zk, SiteConfiguration siteConf, String... args) {
```

Comment:
The two use cases should be:

1. ZooZap.execute(mainArgs), or
2. ZooZap.zap(serverContext, options)

The execute version should do the options parsing, and the other should just be able to be called directly, with options suitable for code (like a Set of services to zap, rather than option strings like `-manager`).